token-lending/program/src/processor.rs

```diff
@@ -1528,18 +1473,137 @@ fn process_set_lending_market_owner(
+        collateral.market_value = market_value;
+
+        let loan_to_value_rate = Rate::from_percent(deposit_reserve.config.loan_to_value_ratio);
+        let liquidation_threshold_rate =
+            Rate::from_percent(deposit_reserve.config.liquidation_threshold);
+
+        deposited_value = deposited_value.try_add(market_value)?;
+        loan_to_value_ratio =
+            loan_to_value_ratio.try_add(market_value.try_mul(loan_to_value_rate)?)?;
+        liquidation_threshold =
+            liquidation_threshold.try_add(market_value.try_mul(liquidation_threshold_rate)?)?;
+    }
+
+    for (index, liquidity) in obligation.borrows.iter_mut().enumerate() {
+        let borrow_reserve_info = next_account_info(account_info_iter)?;
+        if borrow_reserve_info.owner != program_id {
+            msg!(
+                "Borrow reserve provided for liquidity {} is not owned by the lending program",
+                index
+            );
+            return Err(LendingError::InvalidAccountOwner.into());
+        }
+        if liquidity.borrow_reserve != *borrow_reserve_info.key {
+            msg!(
+                "Borrow reserve of liquidity {} does not match the borrow reserve provided",
+                index
+            );
+            return Err(LendingError::InvalidAccountInput.into());
+        }
+
+        let borrow_reserve = Reserve::unpack(&borrow_reserve_info.data.borrow())?;
+        if borrow_reserve.last_update.is_stale(clock.slot)? {
+            msg!(
+                "Borrow reserve provided for liquidity {} is stale and must be refreshed in the current slot",
+                index
+            );
+            return Err(LendingError::ReserveStale.into());
+        }
+
+        let decimals = 10u64
+            .checked_pow(borrow_reserve.liquidity.mint_decimals as u32)
+            .ok_or(LendingError::MathOverflow)?;
+
+        liquidity.accrue_interest(borrow_reserve.liquidity.cumulative_borrow_rate_wads)?;
+        let market_value = liquidity
+            .borrowed_amount_wads
+            .try_mul(borrow_reserve.liquidity.market_price)?
+            .try_div(decimals)?;
+        liquidity.market_value = market_value;
+        borrowed_value = borrowed_value.try_add(market_value)?;
+    }
+
+    if account_info_iter.peek().is_some() {
+        msg!("Too many obligation deposit or borrow reserves provided");
+        return Err(LendingError::InvalidAccountInput.into());
+    }
+
+    obligation.deposited_value = deposited_value;
+    obligation.borrowed_value = borrowed_value;
+
+    if deposited_value == Decimal::zero() {
+        obligation.loan_to_value_ratio = Rate::zero();
+        obligation.liquidation_threshold = Rate::zero();
     } else {
-        Ok(())
+        obligation.loan_to_value_ratio =
+            Rate::try_from(loan_to_value_ratio.try_div(deposited_value)?)?;
+        obligation.liquidation_threshold =
+            Rate::try_from(liquidation_threshold.try_div(deposited_value)?)?;
```

Comment:
Hmm. Maximum allowed borrow value is zero at the weighted average LTV, and effectively "negative" above it. The liquidation threshold can't be inferred directly from that.

Reply:
I didn't mean maximum *additional* allowed borrow value, I was thinking it could represent the total allowed, not counting what is already borrowed.

Reply:
I'm still unclear on this.

> You could store the liquidation threshold as the maximum allowed borrow value

This is effectively what `loan_to_value_ratio` represents here. Liquidation threshold can't be inferred from maximum borrows, because you are not allowed to borrow above the LTV ratio.

I also think as a base protocol, it makes sense to keep these values the same to those used for the reserves so that a protocol built on top can easily reason about the relationship.

Reply:
I was just suggesting you keep everything the same except skip the division which could be done later since we start the deposited value anyways

Reply:
I see. I'll take a look at where this is used and test that out.

Reply:
Done in b3a584eb8d6ba13fb06f780c9c4847c0ef9cefd1 and 187a21a2fabed9d05ea33c8bdc653bb5187d0882